lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,996 @@
+  login_password:
+    description:
+    - Password used to authenticate with PostgreSQL.
+    type: str
+  login_host:
+    description:
+    - Host running PostgreSQL.
+    type: str
+  login_unix_socket:
+    description:
+    - Path to a Unix domain socket for local connections.
+    type: str
+  ssl_mode:
+    description:
+    - Determines whether or with what priority a secure SSL TCP/IP connection
+      will be negotiated with the server.
+    - See U(https://www.postgresql.org/docs/current/static/libpq-ssl.html) for
+      more information on the modes.
+    - Default of C(prefer) matches libpq default.
+    type: str
+    choices: [ allow, disable, prefer, require, verify-ca, verify-full ]
+    default: prefer
+  ssl_rootcert:
+    description:
+    - Specifies the name of a file containing SSL certificate authority (CA)
+      certificate(s).
+    - If the file exists, the server's certificate will be
+      verified to be signed by one of these authorities.
+    type: str
+notes:
+- The default authentication assumes that you are either logging in as or
+  sudo'ing to the postgres account on the host.
+- To avoid "Peer authentication failed for user postgres" error,
+  use postgres user as a I(become_user).
+- This module uses psycopg2, a Python PostgreSQL database adapter. You must
+  ensure that psycopg2 is installed on the host before using this module. If
+  the remote host is the PostgreSQL server (which is the default case), then
+  PostgreSQL must also be installed on the remote host. For Ubuntu-based
+  systems, install the postgresql, libpq-dev, and python-psycopg2 packages
+  on the remote host before using this module.
+requirements: [ psycopg2 ]
+author:
+- Andrew Klychkov (@Andersson007)
+'''
+
+EXAMPLES = r'''
+# Display facts from postgres hosts.
+# ansible postgres -m postgresql_facts
+
+# Display only databases and roles facts from all hosts using shell-style wildcards:
+# ansible all -m postgresql_facts -a 'filter=dat*,rol*'
+
+# Display only replications and repl_slots facts from standby hosts using shell-style wildcards:
+# ansible standby -m postgresql_facts -a 'filter=repl*'
+
+# Display all facts from databases hosts except settings:
+# ansible databases -m postgresql_facts -a 'filter=!settings'
+
+- name: Collect PostgreSQL version and extensions
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter: ver*,ext*
+
+- name: Collect all facts except settings and roles
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter: "!settings,!roles"
+
+# On FreeBSD with PostgreSQL 9.5 version and lower use pgsql user to become
+# and pass "postgresq" as a database to connect to
+- name: Collect tablespaces and repl_slots facts
+  become: yes
+  become_user: pgsql
+  postgresql_facts:
+    db: postgres
+    filter:
+    - tablesp*
+    - repl_sl*
+
+- name: Collect all facts except settings
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter:
+    - "!settings"
```

Comment:
Example with `!settings` is already present on line 122. Would be better to not repeat the same facts